lib/trino-hdfs/src/main/java/io/trino/hdfs/HdfsModule.java

```diff
@@ -33,6 +33,7 @@ public void configure(Binder binder)
 
         binder.bind(HdfsConfigurationInitializer.class).in(Scopes.SINGLETON);
         newSetBinder(binder, ConfigurationInitializer.class);
-        newSetBinder(binder, DynamicConfigurationProvider.class);
+        newSetBinder(binder, DynamicConfigurationProvider.class).addBinding()
+                .to(HdfsConfigurationProvider.class).in(Scopes.SINGLETON);
```

Comment:
`.addBinding().to(...)` typically goes on one line

```suggestion
        newSetBinder(binder, DynamicConfigurationProvider.class)
                                        addBinding().to(HdfsConfigurationProvider.class).in(Scopes.SINGLETON);
```